dex/networks/btc/script.go

```diff
@@ -301,32 +301,24 @@ func ParseScriptType(pkScript, redeemScript []byte) BTCScriptType {
 // MakeContract creates a segwit atomic swap contract. The secretHash MUST
 // be computed from a secret of length SecretKeySize bytes or the resulting
 // contract will be invalid.
-func MakeContract(recipient, sender string, secretHash []byte, lockTime int64, segwit bool, chainParams *chaincfg.Params) ([]byte, error) {
-	rAddr, err := btcutil.DecodeAddress(recipient, chainParams)
-	if err != nil {
-		return nil, fmt.Errorf("error decoding recipient address %s: %w", recipient, err)
-	}
-	sAddr, err := btcutil.DecodeAddress(sender, chainParams)
-	if err != nil {
-		return nil, fmt.Errorf("error decoding sender address %s: %w", sender, err)
-	}
+func MakeContract(rAddr, sAddr btcutil.Address, secretHash []byte, lockTime int64, segwit bool, chainParams *chaincfg.Params) ([]byte, error) {
```

Comment:
Changing these arguments from `string` to `btcutil.Address` was needed to make things work with the new customizable `AddressDecoder`. It also appears to be a simplification when the changes are taken as a whole.